src/fileio.cpp

```diff
@@ -261,10 +261,18 @@ bool RcdFileReader::GetBlob(void *address, size_t length)
  * @param path Path of the directory.
  * @todo At the time of writing (2021-06-30) this is tested only on Linux. Before using it anywhere else, test this on all platforms (especially Windows).
  */
-void MakeDirectory(const std::string &path)
+void MakeDirectory(std::string path)
 {
-	if (PathIsDirectory(path.c_str())) return;
+	if (path.empty() || PathIsDirectory(path.c_str())) return;
 
+	/* Strip trailing path separators. */
+	while (StrEndsWith(path.c_str(), DIR_SEP, false)) {
+		for (size_t i = strlen(DIR_SEP); i > 0; --i) {
+			path.pop_back();
+		}
+	}
```

Comment:
Perhaps keep at least one character? `/////` as path isn't very sane, but at least `/` is then a technically valid end result rather than the empty path (which I am not sure what that would mean here).

Reply:
This should already be covered by the `PathIsDirectory()` call in line 266 – I think we can always guarantee that the root directory exists.

Reply:
ok